prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/notification/NotificationWorkFlowTask.java

```diff
@@ -0,0 +1,103 @@
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import org.springframework.http.HttpHeaders;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.*;
+
+@Slf4j
+public class NotificationWorkFlowTask extends BaseWorkFlowTask {
+
+	private final NotificationMessageApi apiInstance;
+
+	public NotificationWorkFlowTask(String basePath) {
+		ApiClient apiClient = Configuration.getDefaultApiClient();
+		apiClient.addDefaultHeader(HttpHeaders.AUTHORIZATION, "Basic " + CredUtils.getBase64Creds("test", "test"));
+		apiClient.setBasePath(basePath); // basePath = "http://localhost:8080"
+		this.apiInstance = new NotificationMessageApi(apiClient);
+	}
+
+	@Override
+	public @NonNull List<WorkFlowTaskParameter> getWorkFlowTaskParameters() {
+		LinkedList<WorkFlowTaskParameter> params = new LinkedList<>();
+		params.add(WorkFlowTaskParameter.builder().key("type").type(WorkFlowTaskParameterType.TEXT).optional(false)
+				.description("Message Type").build());
+		params.add(WorkFlowTaskParameter.builder().key("body").type(WorkFlowTaskParameterType.TEXT).optional(false)
+				.description("Message Body").build());
+		params.add(WorkFlowTaskParameter.builder().key("subject").type(WorkFlowTaskParameterType.TEXT).optional(false)
+				.description("Message Subject").build());
+		// TODO Add an option for List parameter
+		params.add(WorkFlowTaskParameter.builder().key("userNames").type(WorkFlowTaskParameterType.TEXT).optional(true)
+				.description("Comma separated list of user names").build());
+		params.add(WorkFlowTaskParameter.builder().key("groupNames").type(WorkFlowTaskParameterType.TEXT).optional(true)
+				.description("Comma separated list of group names").build());
+		return params;
+	}
+
+	@Override
+	public @NonNull List<WorkFlowTaskOutput> getWorkFlowTaskOutputs() {
+		return super.getWorkFlowTaskOutputs();
+	}
+
+	@Override
+	public HashMap<String, Map<String, Object>> getAsJsonSchema() {
+		return super.getAsJsonSchema();
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		NotificationMessageCreateRequestDTO notificationMessageCreateRequestDTO = new NotificationMessageCreateRequestDTO(); // NotificationMessageCreateRequestDTO
+																																// |
+
+		try {
+			notificationMessageCreateRequestDTO.messageType(getRequiredParameterValue(workContext, "type"));
+			notificationMessageCreateRequestDTO.body(getRequiredParameterValue(workContext, "body"));
+			notificationMessageCreateRequestDTO.subject(getRequiredParameterValue(workContext, "subject"));
+			List<String> userNames = toList(getOptionalParameterValue(workContext, "userNames", null));
+			List<String> groupNames = toList(getOptionalParameterValue(workContext, "groupNames", null));
+			if (userNames == null && groupNames == null) {
```

Comment:
Is there the need to check is the lists are empty as well? Or just the `null` check is ok

Reply:
Added also "isEmpty" validation
Thanks